tests/integration/helpers.py

```diff
@@ -18,3 +27,62 @@ async def get_leader_unit(model, application_name: str) -> Unit:
         if unit.application == application_name and await unit.is_leader_from_status():
             return unit
     raise RuntimeError(f"Leader unit for `{application_name}` not found.")
+
+
+async def get_unit_status_messages(
+    ops_test: OpsTest, app_name: str = APP_NAME
+) -> List[tuple[str, str]]:
+    """Get the status messages from all the units of the given application."""
+    return_code, stdout, stderr = await ops_test.juju("status", "--format", "yaml", app_name)
+    if return_code:
+        raise RuntimeError(stderr)
+    output = yaml.safe_load(stdout)
+    unit_statuses = output["applications"][app_name]["units"]
+    print(unit_statuses)
+    return [
+        (unit_name, unit_status["workload-status"].get("message", ""))
+        for (unit_name, unit_status) in unit_statuses.items()
+    ]
+
+
+async def wait_for_status_message(
+    ops_test: OpsTest,
+    expected_message: str,
+    app_name: str = APP_NAME,
+    count: int = 1,
+    timeout: int = 100,
+    cadence: int = 2,
+) -> None:
+    """Wait for the correct status messages to appear.
+
+    Args:
+        ops_test: Ops test Framework.
+        app_name: Application name of the Vault, defaults to "vault-k8s"
+        count: How many units are expected to be emitting the message
+        expected_message: The message that vault units should be setting as a status message
+        timeout: Wait time, in seconds, before giving up
+        cadence: How often to check the status of the units
+
+    Raises:
+        TimeoutError: If the expected amount of statuses weren't found in the given timeout.
+    """
+    seen = 0
+    unit_statuses = []
+    while timeout > 0:
+        unit_statuses = await get_unit_status_messages(ops_test, app_name=app_name)
+        seen = 0
+        for unit_name, unit_status_message in unit_statuses:
+            if unit_status_message == expected_message:
+                seen += 1
+
+        if seen == count:
+            return
+        time.sleep(cadence)
+        timeout -= cadence
+    import pdb
+
+    pdb.set_trace()
```

Comment:
Also missed this when carrying over from test branch. Need to remove.

Reply:
https://github.com/canonical/vault-k8s-operator/pull/429/commits/e877ef39c6ba4eb4b2d8d5f740f2e2aafb6a92b2